app/dao/users_dao.py

```diff
@@ -229,7 +251,9 @@ def get_user_and_accounts(user_id):
         joinedload('organisations'),
         joinedload('organisations.services'),
         joinedload('services.organisation'),
-    ).one()
+    )
+
+    return db.session.scalars(stmt).unique().one()
```

Comment:
Ugg.  This isn't unit tested.

Reply:
I verified that the sql is the same before and after this change, and I think I finally have my head around this.  According to the sqlalchemy [docs on joinedload](https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html#sqlalchemy.orm.joinedload), the join defaults to left-outer-join, but it can also be an inner-join.

According to the [tutorial on w3schools](https://www.w3schools.com/SQL/sql_join.asp):
- LEFT (OUTER) JOIN: Returns all records from the left table, and the matched records from the right table
- (INNER) JOIN: Returns records that have matching values in both tables

Noting that this function returns a User instance, the inner join is not appropriate because it's possible that a user exists who is not assigned to any services or organizations.

That leaves left-outer-join, which first returns all User instances before filtering with the "where" clause.  (I verified that the generated sql is the same rather or not the "where" clause precedes the "options" clause.)  Because these are many-to-many relationships, you could have rows with duplicate User primary key values.  So which row would you return?  The "where" clause filters by primary key, but that still might leave multiple, duplicate rows.  So use "unique".